php/src/Google/Protobuf/Internal/DescriptorProto.php

```diff
@@ -307,25 +330,24 @@ public function hasReservedRange()
     }
 
     /**
-     * <pre>
      * Reserved field names, which may not be used by fields in the same message.
      * A given name may only be reserved once.
-     * </pre>
      *
-     * <code>repeated string reserved_name = 10;</code>
+     * Generated from protobuf field <code>repeated string reserved_name = 10;</code>
+     * @return \Google\Protobuf\Internal\RepeatedField
      */
     public function getReservedName()
     {
         return $this->reserved_name;
     }
 
     /**
-     * <pre>
      * Reserved field names, which may not be used by fields in the same message.
      * A given name may only be reserved once.
-     * </pre>
      *
-     * <code>repeated string reserved_name = 10;</code>
+     * Generated from protobuf field <code>repeated string reserved_name = 10;</code>
+     * @param array|\Google\Protobuf\Internal\RepeatedField $var
```

Comment:
@bshaffer @TeBoring What do you think about replacing `array|\Google\Protobuf\Internal\RepeatedField` with `string[]|\Google\Protobuf\Internal\RepeatedField`? This is more descriptive, since it accepts specifically an array of strings.

Similarly for other types, we could generate `\Fully\Qualified\Type[]|\Google\Protobuf\DescriptorProto[]`

This is longer, but means that the type to be used is available